### PR TITLE
Add the ability to DM user on punishment + Improve commands

### PR DESCRIPTION
**Pull request by Jeymen**
I've added the ability to DM user on punishment. If the user's DM's are off, a follow-up message will be sent alerting the moderators. I've also improved the commands.

### CODE REVIEW

File: src/commands/ban.ts [hunk 1]
```diff
@@ -47,6 +47,8 @@ export class ban {
       try {
         await guild.members.ban(user, { reason: reason });
         await interaction.editReply("✅ User has been banned.");
+        await user.send(`You have been banned for ${reason}.`)
+          .catch(() => interaction.followUp("Can't send DM to the user!"));
```
Comment (Pdzly) [inline src/commands/ban.ts:51]: I would add here from whom for auditability

File: src/commands/ban.ts [hunk 2]
```diff
@@ -81,6 +83,8 @@ export class ban {
       try {
         await guild.members.unban(user);
         await interaction.editReply("✅ User has been unbanned.");
+        await user.send(`You have been unbanned.`)
```
Comment (Pdzly) [inline src/commands/ban.ts:86]: Same here

File: src/commands/kick.ts [hunk 3]
```diff
@@ -47,6 +47,8 @@ export class kick {
       try {
         await guild.members.kick(user, reason);
         await interaction.editReply("✅ User has been kicked.");
+        await user.send(`You have been kicked for ${reason}.`)
```
Comment (Pdzly) [inline src/commands/kick.ts:50]: Same here

File: src/commands/mute.ts [hunk 4]
```diff
@@ -57,6 +57,9 @@ export class MuteCommand {
             await user.timeout(timeoutTime);
             await interaction.editReply(`${user} has been muted for ${time} minute(s) for ${reason}.`);
 
+            await user.send(`You have been muted for ${time} minute(s) for ${reason}`)
```
Comment (Pdzly) [inline src/commands/mute.ts:60]: Sane here

File: src/commands/kick.ts [hunk 5]
```diff
@@ -30,29 +29,30 @@ export class kick {
...
+      await interaction.editReply("You cannot kick yourself!");
+      return;
     }
 
-    const guild: Guild = interaction.guild;
-    await interaction.deferReply();
+    try {
+      await user.send(`You have been kicked for ${reason} in ${interaction.guild.name}`)
+        .catch(() =>
+          interaction.followUp(`Can't send DM to ${user}!`)
```
Comment (Pdzly) [inline src/commands/kick.ts:46]: Same here!

File: src/commands/mute.ts [hunk 6]
```diff
@@ -92,9 +96,11 @@ export class MuteCommand {
         try {
             await user.timeout(null);
             await interaction.editReply(`${user} has been unmuted.`);
+            await user.send(`You have been unmuted in ${interaction.guild.name}`)
```
Comment (Pdzly) [inline src/commands/mute.ts:99]: Here too

**Review by Pdzly — CHANGES_REQUESTED** [review]

**Comment by Jeymen** [comment]
This is being worked on by @Jeymen.

**Review by Pdzly — APPROVED** [review]
Nice work!  The only issue is still the permissions ( is the user over the other user )

**Comment by Jeymen** [comment]
> Nice work! The only issue is still the permissions ( is the user over the other user )

That will be worked on in a future pull request!

**Pull request merged**